File: 110.BalancedBinaryTree/BalancedBinaryTree.java
public class BalancedBinaryTree {

    public static void main(String[] args) {
	// write your code here
    }

    public boolean isBalanced(TreeNode root) {
        if(root == null){
            return true;
        }
        if(Math.abs(height(root.left) - height(root.right)) <= 1){
            return isBalanced(root.left) && isBalanced(root.right);
        }else{
            return false;
        }
    }

    public int height(TreeNode node){
        if(node == null){
            return -1;
        }else{
            return Math.max(height(node.left), height(node.right)) + 1;
        }
    }
}

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

File: LeetcodeQuestion/110.BalancedBinaryTree/BalancedBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def isBalanced(self, root: TreeNode) -> bool:
        if root == None:
            return True
        
        if abs(height(root.left) - height(root.right)) <= 1:
            return Solution.isBalanced(self, root.left) & Solution.isBalanced(self, root.right)
        else:
            return False
    
def height(root):
    if root == None:
        return -1;
    if root.left == None and root.right is None:
        return 0
    else:
        return max(height(root.left), height(root.right)) + 1